File: unity/BlackjackUnity/Assets/SoundEffects.cs
using UnityEngine;
using System.Collections.Generic;

public class SoundEffects {
    private bool soundOn;
    private AudioSource source;
    private List<string> strSfx;
    private List<AudioClip> sfx;

    /// <summary>
    /// Sound effects implements a mechanism for playing sound effects.
    /// </summary>
    /// <param name="strSfx">List of sound effect names.</param>
    /// <param name="sfx">List of sound effect objects.</param>
    public SoundEffects(List<string> strSfx, List<AudioClip> sfx) {
        soundOn = true;
        this.source = new AudioSource();
        this.strSfx = strSfx;
        this.sfx = sfx;
        Debugger.Emit(true, strSfx.Count);
        Debugger.Emit(true, sfx.Count);
    }

    /// <summary>
    /// Toggle sound effects on/off.
    /// </summary>
    /// <returns>Are sound effects on?</returns>
    public bool Toggle() {
        if (soundOn) soundOn = false;
        else soundOn = true;
        return soundOn;
    }

    /// <summary>
    /// Play a sound effect.
    /// </summary>
    /// <param name="effect">Name of sound effect to play.</param>
    public void Play(string effect) {
        AudioClip audio = sfx[1]; //sfx[strSfx.IndexOf(effect)];
        Debugger.Emit(true, sfx[1].ToString());
        source.clip = audio;
        if(soundOn) source.Play();
    }
}

File: java/SoundEffects.java
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 * 
 * Java Swing/AWT implementation.
*/

import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.io.IOException;
import org.newdawn.easyogg.OggClip;

public class SoundEffects
{
    private List<String> strEffects;
    private List<OggClip> effects;
    private boolean soundOn;
    
    /**
     * SoundEffects implements playing game sound effects.
    */
    public SoundEffects()
    {
        String s = "sounds/";
        String[] names = { "deal", "shuffle", "hit", "reveal" };
        strEffects = Arrays.asList(names);
        try
        {
            effects = new ArrayList<OggClip>();
            for(int i = 0; i < strEffects.size(); i++)
            {
                effects.add(new OggClip(String.format("%s/%s.ogg", s, strEffects.get(i))));
            }
            soundOn = true;
        }
        catch(IOException e)
        {
            e.printStackTrace();
        }
    }
    
    /**
     * Toggle sound effects on/off.
     * @return Are sound effects on?
    */
    public boolean toggle()
    {
        if(soundOn) soundOn = false;
        else soundOn = true;
        return soundOn;
    }
    
    /**
     * Play a sound effect.
     * @param effect Name of sound effect to play.
    */
    public void play(String effect)
    {
        if(soundOn)
        {
            effects.get(strEffects.indexOf(effect)).play();
        }
    }
}
